nixos/modules/services/video/mirakurun.nix

```diff
@@ -0,0 +1,155 @@
+            automatically be generated at runtime.
+          '';
+        };
+
+        channelSettings = mkOption {
+          type = with types; nullOr settingsFmt.type;
+          default = null;
+          example = literalExample ''
+            [
+              {
+                name = "channel";
+                types = "GR";
+                channel = "0";
+              }
+            ];
+          '';
+          description = ''
+            Options which are added to channels.yml. If none is specified, it
+            will automatically be generated at runtime.
+          '';
+        };
+      };
+    };
+
+    config = mkIf cfg.enable {
+      environment.systemPackages = [ mirakurun ];
+      environment.etc = {
+        "mirakurun/server.yml".source = settingsFmt.generate "server.yml" cfg.serverSettings;
+        "mirakurun/tuners.yml" = mkIf (cfg.tunerSettings != null) {
+          source = settingsFmt.generate "tuners.yml" cfg.tunerSettings;
+          mode = "0644";
+          user = username;
+          group = groupname;
+        };
+        "mirakurun/channels.yml" = mkIf (cfg.channelSettings != null) {
+          source = settingsFmt.generate "channels.yml" cfg.channelSettings;
+          mode = "0644";
+          user = username;
+          group = groupname;
+        };
+      };
+
+      users.users.mirakurun = {
+        description = "Mirakurun user";
+        group = "video";
+        isSystemUser = true;
+      };
+
+      services.mirakurun.serverSettings = {
+        inherit (cfg) logLevel;
+        path = mkIf (cfg.domainSocket != null) cfg.domainSocket;
+        port = mkIf (cfg.port != null) cfg.port;
+      };
+
+      systemd.services.mirakurun = {
+        description = mirakurun.meta.description;
+        wantedBy = [ "multi-user.target" ];
+        after = [ "network.target" ];
+        serviceConfig = {
+          ExecStart = "${mirakurun}/bin/mirakurun";
+          User = username;
+          Group = groupname;
+          RuntimeDirectory="mirakurun";
+          PermissionsStartOnly = true;
+          Nice = -10;
+          IOSchedulingClass = "realtime";
+          IOSchedulingPriority = 7;
+        };
+
+        environment = {
+          SERVER_CONFIG_PATH = "/etc/mirakurun/server.yml";
+          TUNERS_CONFIG_PATH = "/etc/mirakurun/tuners.yml";
+          CHANNELS_CONFIG_PATH = "/etc/mirakurun/channels.yml";
+          SERVICES_DB_PATH = "/var/db/mirakurun/services.json";
+          PROGRAMS_DB_PATH = "/var/db/mirakurun/programs.json";
+          NODE_ENV = "production";
+        };
+
+        preStart = ''
+          confdir=/etc/mirakurun
+          dbdir=/var/db/mirakurun
+
+          if [ ! -d "$dbdir" ]; then
+            mkdir -p "$dbdir"
+          fi
+
+          # confdir needs to be writable so that mirakurun can autogenerate
+          # channel/tuner configuration if one doesn't exist
+          for dir in "$confdir" "$dbdir"; do
+            chown -R "${username}:${groupname}" "$dir"
```

Comment:
The database directory could be configured with `StateDirectory` (`man systemd.exec`) instead of this prestart script. And the conf dir could be done with the `systemd.tmpfiles.rules` option.

Reply:
Fixed